packages/assets-controllers/CHANGELOG.md

```diff
@@ -5,6 +5,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- `TokenListController` now exports a `TokenListControllerMessenger` type ([#3609](https://github.com/MetaMask/core/pull/3609)).
+- `TokenDetectionController` exports types `TokenDetectionControllerMessenger`, `TokenDetectionControllerActions`, `TokenDetectionControllerGetStateAction`, `TokenDetectionControllerEvents`, `TokenDetectionControllerStateChangeEvent` ([#3609](https://github.com/MetaMask/core/pull/3609)).
+
+### Changed
+- **BREAKING:** `TokenDetectionController` is upgraded to extend `BaseControllerV2` and `StaticIntervalPollingController` ([#3609](https://github.com/MetaMask/core/pull/3609)).
+  - The constructor now expects an options object as its only argument, with required properties `messenger`, `networkClientId`, required callbacks `onPreferencesStateChange`, `getBalancesInSingleCall`, `getPreferencesState`, and optional properties `interval`, `selectedAddress`. Note that the `config` object is no longer used by the constructor.
+  - Polling can only be initiated by calling `start` or `startPollingByNetworkClientId` from a controller client. There is no longer an option to automatically start polling when the controller is instantiated.
```

Comment:
Which option was this?

Reply:
Ah, I used "polling" here to include passively calling `detectTokens`, which is definitely confusing. Also, the disabled option was previously true by default, which makes the current API a breaking change. Thoughts on the following? 

```suggestion
  - **BREAKING:** By default, the controller may now passively call the `detectTokens` method in response to subscribed events. To disable all network requests, initialize the controller with the `disabled` option set to true, or call the `stop` method.
```

I guess whether to keep this change would depend on the expected behavior of polling controllers upon initialization. If we're going to use a dedicated service for network calls and let it handle blocking outgoing requests, perhaps it makes sense to enable passive updates in the polling controller itself? 

But if this is disruptive to the current implementation I can revert the default value for `disabled`. It will require some changes in the tests, probably including restoring these `start` calls: 73eccbf25f42f340cbac0bb7892a3d32fce8b7aa.

Reply:
I'd prefer that we keep the current default, and consider changing this independently even if we do want to. Better to minimize scope of we can.

Reply:
I see what you mean, this is a bit tricky. I guess this dilemma was introduced with these new polling functions, but made worse with this change because runtime config modification is no longer possible.

For the `CurrencyRateController`, the DevEx team decided to deal with this by calling `stopAllPolling` when the UI closes. So they didn't need to use the disabled state for the newer polling API. But this is problematic because now those polls won't be going anymore when the UI re-opens, even if they might still be needed.

The easiest solution would be to make it possible to update `disabled` at runtime. I don't know what you meant about that seeming unsafe, nor do I understand what you're referring to by "not in keeping with the idea of having this be controlled by a network service".

Longer-term I'd like us to take these services out of the controller altogether, so that there is never a need to disable the controller. @mcmire and I have also discussed adding a "pause" method to the polling controller, so that they can be paused and unpaused without being deregistered (this idea was from before the service refactor was discussed, I am unsure if we still need this). But both of those ideas seem fairly substantial, I am guessing that we'd be better off trying to land this first, now that this work is already done.

Reply:
I was referring to the longer-term solution you mentioned, but I can see that trying to anticipate those changes is premature at this point. 

I added a `disable` and `enable` method for now: 05cf2bd3089d2b58b33bcb21ae3bed6f909c4f6b. Does this seem like a good approach?

Reply:
Looks reasonable to me!

Reply:
Could use additional tests for those two new methods though